bundles/nl.esi.comma.expressions/src/nl/esi/comma/expressions/validation/ExpressionValidator.xtend

```diff
@@ -203,6 +203,13 @@ class ExpressionValidator extends AbstractExpressionValidator {
 	@Check
 	def checkTypingExpression(Expression e){
 		switch(e){
+		    ExpressionEqual:{
+		        val leftType = e.left.typeOf
+		        val rightType = e.right.typeOf
+                if(leftType != rightType){
```

Comment:
This for sure is not correct. Please look at other code where functions like `synonym`, `identical` or `subTypeOf` are being used.

@dbera and @samira-shirzadeh: I think that before implementing this, it should be decided was equivalism means in our expression language. Is it by value or also by type? I mean, can an instance of a type that is **based on** a string be compared with another instance of type `string` itself?

I guess that https://github.com/dbera/OfflineMBT/blob/main/bundles/nl.esi.comma.expressions/src/nl/esi/comma/expressions/validation/ExpressionValidator.xtend#L247-L254 already provides a good first attempt as implementation for `ExpressionEqual`